src/main/java/org/broadinstitute/hellbender/tools/copynumber/gcnv/GenerateVCFFromPosteriors.java

```diff
@@ -0,0 +1,224 @@
+            }
+
+            final List<CopyNumberPosteriorLocatableRecord> copyNumberPosteriorLocatableRecordsList =
+                    chunkedCopyNumberPosteriorFileInfo.getLeft();
+
+            gcnvPostProcessor.writeChunkedVariantContext(copyNumberPosteriorLocatableRecordsList, variantPrefix);
+            currentChunk++;
+        }
+        outputWriter.close();
+        return;
+    }
+
+    /**
+     * Read in a single posterior chunked file
+     */
+    private Triple<List<CopyNumberPosteriorLocatableRecord>, IntegerCopyNumberStateCollection, String> readChunkedPosteriorFileFromDirectory(final File chunkRootDirectory,
+                                                                                                                                             final SAMSequenceDictionary samSequenceDictionary) {
+        //get a list of intervals associated with chunk currently being processed
+        final SimpleIntervalCollection chunkSimpleIntervalCollection =
+                new SimpleIntervalCollection(new File(getIntervalFileFromChunkDirectory(chunkRootDirectory.getAbsolutePath()).getAbsolutePath()));
+
+        //check that sequence dictionaries are equivalent among all chunks
+        if (!chunkSimpleIntervalCollection.getMetadata().getSequenceDictionary().equals(samSequenceDictionary)) {
+            throw new UserException.BadInput("Not all sequence dictionaries are equivalent among interval lists in chunk directories");
+        }
+        final File chunkPosteriorFile = getPosteriorFileFromChunkDirectory(chunkRootDirectory.getAbsolutePath(), sampleDirectoryName);
+
+        final List<String> copyNumberStateColumns = getPosteriorFileColumns(chunkPosteriorFile);
+        final IntegerCopyNumberStateCollection chunkCopyNumberStateCollection = new IntegerCopyNumberStateCollection(copyNumberStateColumns);
+
+        final ChunkedCopyNumberPosteriorCollection chunkedCopyNumberPosteriorCollection =
+                new ChunkedCopyNumberPosteriorCollection(chunkPosteriorFile, chunkCopyNumberStateCollection);
+
+        //Combine together chunked interval list and chunked posterior collection into list of CopyNumberPosteriorLocatableRecord
+        final List<CopyNumberPosteriorLocatableRecord> copyNumberPosteriorLocatableRecordList =
+                IntStream.range(0, chunkedCopyNumberPosteriorCollection.size())
+                        .mapToObj(i -> new CopyNumberPosteriorLocatableRecord(chunkSimpleIntervalCollection.getIntervals().get(i),
+                                chunkedCopyNumberPosteriorCollection.getRecords().get(i)))
+                        .collect(Collectors.toList());
+
+        return new ImmutableTriple<>(copyNumberPosteriorLocatableRecordList, chunkCopyNumberStateCollection, chunkedCopyNumberPosteriorCollection.getMetadata().getSampleName());
+    }
+
+    /**
+     * Get list of column names of the copy number posterior file
```

Comment:
`Extracts column names from a copy-number log posterior TSV file.` Maybe also rename `get` to `extract` in the method name.

Reply:
Done